### DISCUSSION TITLE
[@types/yaireo__tagify] issue with imports

### DISCUSSION THREAD

**Discussion by simkimsia** [DefinitelyTyped/DefinitelyTyped#70285] [Issues with a @types package]
https://github.com/yairEO/tagify/issues/1364#issuecomment-2286130016

I came via this comment

Basically the underlying npm package has changed 

I hope that @Brakebein ans @blutorange can help update this

[a comment by an automated account is not shown]

**Comment by simkimsia** [comment]
I tried the following changes

`package.json`

![image](https://github.com/user-attachments/assets/d69bb845-4282-47c9-a98a-5366b9a1bf46)

`yaireo__tagify-react-tests.tsx`

![image](https://github.com/user-attachments/assets/60039cac-0ef8-4f05-81b9-9a7972346a94)


then i run `pnpm test yaireo__tagify`

then i get 

```
Build tools:
- rollup@2.79.1

 (ignoring rules: 'no-resolution', 'cjs-only-exports-default', 'unexpected-module-syntax', 'cjs-resolves-to-esm')

❌ Import resolved to JavaScript files, but no type declarations were found. https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/UntypedResolution.md


"@yaireo/tagify"

node10: 🟢 
node16 (from CJS): ❌ No types
node16 (from ESM): ❌ No types
bundler: ❌ No types

***********************************

"@yaireo/tagify/react"

node10: 🟢 
node16 (from CJS): 🟢 
node16 (from ESM): 🟢 
bundler: 🟢 

***********************************

"@yaireo/tagify/react.tagify"

node10: 🟢 
node16 (from CJS): 🟢 
node16 (from ESM): 🟢 
bundler: 🟢 

***********************************

"@yaireo/tagify/dist/react.tagify"

node10: ❌ No types
node16 (from CJS): ❌ No types
node16 (from ESM): ❌ No types
bundler: ❌ No types

***********************************

"@yaireo/tagify/dist/react.tagify.jsx"

node10: ❌ No types
node16 (from CJS): 🟢 (CJS)
node16 (from ESM): 🟢 (CJS)
bundler: 🟢 

***********************************

"@yaireo/tagify/dist/tagify.min.js"

node10: 🟢 
node16 (from CJS): ❌ No types
node16 (from ESM): ❌ No types
bundler: ❌ No types

***********************************

"@yaireo/tagify/dist/tagify.min"

node10: 🟢 
node16 (from CJS): ❌ No types
node16 (from ESM): ❌ No types
bundler: ❌ No types

***********************************

"@yaireo/tagify/dist/tagify.esm.js"

node10: ❌ No types
node16 (from CJS): ❌ No types
node16 (from ESM): ❌ No types
bundler: ❌ No types

***********************************

"@yaireo/tagify/dist/tagify.polyfills.min.js"

node10: ❌ No types
node16 (from CJS): ❌ No types
node16 (from ESM): ❌ No types
bundler: ❌ No types

***********************************

"@yaireo/tagify/dist/tagify.vue"

node10: 🟢 
node16 (from CJS): 🟢 
node16 (from ESM): 🟢 
bundler: 🟢 

***********************************

"@yaireo/tagify/dist/tagify.css"

node10: 🟢 
node16 (from CJS): 🟢 
node16 (from ESM): 🟢 
bundler: 🟢 

***********************************
    at combineErrorsAndWarnings (/Users/kimsia/Projects/npm-packages/DefinitelyTyped/node_modules/.pnpm/@definitelytyped+dtslint@0.2.22_typescript@5.6.0-dev.20240813/node_modules/@definitelytyped/dtslint/src/index.ts:259:26)
    at runTests (/Users/kimsia/Projects/npm-packages/DefinitelyTyped/node_modules/.pnpm/@definitelytyped+dtslint@0.2.22_typescript@5.6.0-dev.20240813/node_modules/@definitelytyped/dtslint/src/index.ts:250:18)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at main (/Users/kimsia/Projects/npm-packages/DefinitelyTyped/node_modules/.pnpm/@definitelytyped+dtslint@0.2.22_typescript@5.6.0-dev.20240813/node_modules/@definitelytyped/dtslint/src/index.ts:103:22)
 ELIFECYCLE  Test failed. See above for more details.
 ```
 
 I am not sure how to solve. can advise? I'm really a newbie to typescript. I read https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/UntypedResolution.md but i am none the wiser. @blutorange

**Comment by blutorange** [comment]
I started working a bit on this https://github.com/blutorange/DefinitelyTyped/tree/update-tagify-exports

The file structure needs to be adjusted so that it matches all those exports, and also so that node 10 can be supported.

The main TODO is figuring out how best to write a type declaration file for the ESM and CJS version (that's also doesn't break any existing code). And then adding more tests for those different exports.

**Reply by simkimsia** [reply]
how can i help?

Sounds liek the todo are:

- [ ] adjust file structure to match the exports
- [ ] adjust file structure to support node 10
- [ ] write a type declarataion file for ESM
- [ ] write a type declarataion file for CJS
- [ ] add tests for different exports

among this list, is there anything i can help with?

**Reply by blutorange** [reply]
I don't much time today, but tomorrow. Let me take a look at this first. Then we'll see.

**Reply by blutorange** [reply]
I made a PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70299 and also added a few methods that were added to tagify after 4.24

**Reply by simkimsia** [reply]
Thank you @blutorange 

Fingers crossed 🤞 all works well

**Reply by blutorange** [reply]
It's merged now and should be released. Let us know if you have any issues.
